book.md

```diff
@@ -7,16 +7,90 @@ documentclass: report
 papersize: A4
 ...
 
-Introduction
-============
+# Introduction
 
-Memory vulnerability based attacks and mitigations
-==================================================
+Compilers, assemblers and similar tools generate all the binary code that
+processors execute.  It is no surprise then that for security analysis and
+hardening relevant for binary code, these tools have a major role to play.
+Often the only practical way to protect all binaries with a particular security
+hardening method is to let the compiler adapt its automatic code generation.
 
-Physical access side channel attacks
-====================================
+With software security becoming even more important in recent years, it is no
+surprise to see an ever increasing variety of security hardening features and
+mitigations against vulnerabilities implemented in compilers.
 
-Remote access side channel attacks
-==================================
+Indeed, compared to a few decades ago, today's compiler developer is much more
+likely to work on security features, at least some of their time.
 
+Furthermore, with the ever-expanding range of techniques implemented, it has
+become very hard to gain a basic understanding of all security
+features implemented in typical compilers.
+
+This poses a practical problem: compiler developers must be able to work on
+security hardening features, yet it is hard to gain a good basic understanding
+of such compiler features.
+
+This book aims to help developers of code generation tools such as JITs,
+compilers, linkers, assemblers, etc. overcome this.
+
+There is a lot of material that can be found explaining individual
+vulnerabilities or attack vectors. There are also lots of presentations
+explaining specific exploits. But there seems to be a limited set of material
+that gives a structured overview of all vulnerabilities and exploits for which
+a code generator could play a role in protecting against them.
+
+This book aims to provide such a structured, broad overview.  It does not
+necessarily go into full details.  Instead it aims to give a thorough
+description of all relevant high-level aspects of attacks, vulnerabilities,
+mitigations and hardening techniques. For further details, this book provides
+pointers to material with more details on specific techniques.
+
+The purpose of this book is to serve as a guide to every compiler developer
+that needs to learn about software security relevant to compilers.  Even though
+the focus is on compiler developers, we expect that this book will also be
+useful to other people working on low-level software.
+
+
+## How this book is created
+
+The idea for this book emerged out of a frustration of not finding a good
+overview on this topic. Kristof Beyls and Georgia Kouvelli, both compiler
```

Comment:
extra l

Reply:
This should now be fixed. Apologies for the typo!